app/controllers/devise/passwords_controller.rb

```diff
@@ -46,7 +46,12 @@ def update
       respond_with resource, location: after_resetting_password_path_for(resource)
     else
       set_minimum_password_length
-      respond_with resource
+
+      if expired_token_error(resource)
+        redirect_to new_password_path(resource_name), alert: resource.errors.full_messages.join("\n")
```

Comment:
I'm not sure about showing the errors in a flash message. The errors were supposed to be displayed inside `app/views/password/new`, when we call [`devise_error_messages!`](https://github.com/plataformatec/devise/blob/16b3d6d67c7e017d461ea17ed29ea9738dc77e83/app/views/devise/passwords/new.html.erb#L4). Is this not working?

Reply:
How would resource errors be carried through a redirect?

Reply:
@jeremywadsack maybe use specific i18n key for this rather than using `full_messages` because it could contains other error messages and the only reason you would go to `new_password_path` for this case is because your token expired.

So probably change it to use `i18n`.

Reply:
it might be
```
        redirect_to new_password_path(resource_name), alert: resource.errors.full_messages_for(:reset_password_token).join("\n")
```